src/core/annotation.js

```diff
@@ -922,12 +922,17 @@ class WidgetAnnotation extends Annotation {
 
     const localResources = getInheritableProperty({ dict, key: "DR" });
     const acroFormResources = params.acroForm.get("DR");
+    const appearanceResources = this.appearance
+      ? this.appearance.dict.get("Resources")
+      : null;
+
```

Comment:
Do you actually need the ternary expression here?
```javascript
const appearanceResources =
  this.appearance && this.appearance.dict.get("Resources")`;
```
ought to suffice, given that all relevant code already contains `Dict` checks and having `appearanceResources` be `null` instead of `undefined` really shouldn't matter.

Reply:
In the future, when all supported browsers handle optional chaining, this can be simplified further to:
```
const appearanceResources = this.appearance?.dict.get("Resources");
```